src/supervisor/kuberenetes-api-wrappers.ts

```diff
@@ -1,32 +1,40 @@
 import * as http from 'http';
 import * as sleep from 'sleep-promise';
+import { IRequestError } from './types';
 
 export const ATTEMPTS_MAX = 3;
 export const DEFAULT_SLEEP_SEC = 1;
 export const MAX_SLEEP_SEC = 5;
 type IKubernetesApiFunction<ResponseType> = () => Promise<ResponseType>;
 
+function shouldRetryRequest(err: IRequestError, attempt: number): boolean {
+  if (attempt === ATTEMPTS_MAX) {
```

Comment:
should probably consider the case where attempt > ATTEMPTS_MAX now being made possible by the extraction of this logic and "separating" it from the loop